code/game/data_huds.dm

```diff
@@ -575,3 +522,46 @@ Diagnostic HUDs!
 
 #undef CACHED_WIDTH_INDEX
 #undef CACHED_HEIGHT_INDEX
+
+/atom/proc/get_visual_width()
+	var/width = get_cached_width()
+	var/height = get_cached_height()
+	var/scale_list = list(
+		width * transform.a + height * transform.b + transform.c,
+		width * transform.a + transform.c,
+		height * transform.b + transform.c,
+		transform.c
+	)
+	return max(scale_list) - min(scale_list)
+
+/atom/proc/get_visual_height()
+	var/width = get_cached_width()
+	var/height = get_cached_height()
+	var/scale_list = list(
+		width * transform.d + height * transform.e + transform.f,
+		width * transform.d + transform.f,
+		height * transform.e + transform.f,
+		transform.f
+	)
+	return max(scale_list) - min(scale_list)
+
+/atom/proc/adjust_hud_position(image/holder, animate_time = null)
+	if (animate_time)
+		animate(holder, pixel_x = -(get_cached_width() - ICON_SIZE_X) / 2, pixel_y = get_cached_height() - ICON_SIZE_Y, time = animate_time)
+		return
+	holder.pixel_x = -(get_cached_width() - ICON_SIZE_X) / 2
```

Comment:
```suggestion
		animate(holder, pixel_x = -(get_cached_width() - ICON_SIZE_X) * 0.5, pixel_y = get_cached_height() - ICON_SIZE_Y, time = animate_time)
		return
	holder.pixel_x = -(get_cached_width() - ICON_SIZE_X) * 0.5
```
micro-op, less cpu cycles yadda yadda

Reply:
Literally noise. Don't seriously tell people this, this is how byond perf myths get spread.

Reply:
We're not doing that, went over this with potato. That's not how this works.

Reply:
I see. I'll keep a memo on that as part of "several performance improvements that don't actually improve anything because it's Byond".

Reply:
> I see. I'll keep a memo on that as part of "several performance improvements that don't actually improve anything because it's Byond".

Multiplication vs division is something you should never tell anyone ever because it does not matter in 2024. Unless you are writing the most optimized of raw ASM possible, it will never matter.

Reply:
got it.